src/server/game/Entities/Player/Player.cpp

```diff
@@ -27243,3 +27247,186 @@ uint32 Player::DoRandomRoll(uint32 minimum, uint32 maximum)
 
     return roll;
 }
+
+uint32 Player::PlayScene(uint32 sceneId, Position const* position /*= nullptr*/)
+{
+    SceneTemplate const* sceneTemplate = sObjectMgr->GetSceneTemplate(sceneId);
+    return PlaySceneByTemplate(sceneTemplate);
+}
+
+uint32 Player::PlaySceneByTemplate(SceneTemplate const* sceneTemplate, Position const* position/* = nullptr*/)
+{
+    if (sceneTemplate == nullptr)
+        return 0;
+
+    SceneScriptPackageEntry const* entry = sSceneScriptPackageStore.LookupEntry(sceneTemplate->scenePackageId);
+
+    if (entry == nullptr)
+        return 0;
+
+    // By default, take player position
+    if (position == nullptr)
+    {
+        Position loc = GetPosition();
+        position = &loc;
```

Comment:
maybe it's better to learn C++ a bit more before working on a complicated project like TC, so many bad things might still build and yet cause lot of troubles at runtime.